scripts/create-darwin-volume.sh

```diff
@@ -83,6 +61,58 @@ suggest_report_error(){
     echo "       please report this @ https://github.com/nixos/nix/issues" >&2
 }
 
+generate_mount_command(){
+    if test_filevault_in_use; then
+        printf "    <string>%s</string>\n" /bin/sh -c '/usr/bin/security find-generic-password -a "Nix" -w | /usr/sbin/diskutil apfs unlockVolume "Nix" -mountpoint /nix -stdinpassphrase'
```

Comment:
Since we're passing the `-s` flag with the UUID when creating the password, we need to be passing it here too, as mentioned [in this comment](https://github.com/NixOS/nix/pull/4181/files#r512356331).

Reply:
This is still on my radar. I set out trying to iron out some of these issues, and found myself feeling like it would be a big leg up to have access to the UI/X affordances of the multi-user installer before tackling them. 

The draft refactor to get that functionality also put me really close (I feel like) to laying some groundwork for directly curing more issues instead of just throwing instructions at the user. In turn, that put me really close to laying some groundwork for uninstalling bits of this.

It isn't a full solution yet (i.e., not covering components I didn't add yet; not covering linux), nor am I sure if it'll end up in scope here or if parts will need to get broken off...